### ISSUE TITLE
GLPK_MI interface errors

### ISSUE THREAD

**Issue by wkschwartz** [cvxgrp/cvxpy#812]
I was going to try to give a minimal test case to reproduce a traceback from CVXPY 1.0.25, GLPK 4.65, and CVXOPT 1.2.3 that ends as follows:
```python-traceback
  ...
  File ".../venv/lib/python3.7/site-packages/cvxpy/problems/problem.py", line 289, in solve
    return solve_func(self, *args, **kwargs)
  File ".../venv/lib/python3.7/site-packages/cvxpy/problems/problem.py", line 571, in _solve
    self, data, warm_start, verbose, kwargs)
  File ".../venv/lib/python3.7/site-packages/cvxpy/reductions/solvers/solving_chain.py", line 194, in solve_via_data
    solver_opts, problem._solver_cache)
  File ".../venv/lib/python3.7/site-packages/cvxpy/reductions/solvers/conic_solvers/glpk_mi_conif.py", line 88, in solve_via_data
    status = self.STATUS_MAP[results_dict['status']]
KeyError: 'infeasible problem'
```

But then I ran into the following, possibly more severe problem.

```python-console
>>> import cvxpy
>>> x = cvxpy.Variable(1, boolean=True)
>>> p = cvxpy.Problem(cvxpy.Maximize(x), [x == 0, x == 1])
>>> p.solve(solver=cvxpy.GLPK_MI)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File ".../venv/lib/python3.7/site-packages/cvxpy/problems/problem.py", line 289, in solve
    return solve_func(self, *args, **kwargs)
  File ".../venv/lib/python3.7/site-packages/cvxpy/problems/problem.py", line 571, in _solve
    self, data, warm_start, verbose, kwargs)
  File ".../venv/lib/python3.7/site-packages/cvxpy/reductions/solvers/solving_chain.py", line 194, in solve_via_data
    solver_opts, problem._solver_cache)
  File ".../venv/lib/python3.7/site-packages/cvxpy/reductions/solvers/conic_solvers/glpk_mi_conif.py", line 75, in solve_via_data
    set(int(i) for i in data[s.BOOL_IDX]))
TypeError: G must be a 'd' matrix
```

**Comment by rileyjmurray** [comment]
I'm not sure if there were dedicated unittests for GLPK in the past. There will be from now on, per PR #910. GLPK_MI is currently passing all tests in that PR. The error you see here is because there are no inequality constraints (and the CVXOPT interface to GLPK assumes > 0 inequality constraints). I've also fixed that error in #910.

**Comment by wkschwartz** [comment]
Does #910 close this?

**Comment by rileyjmurray** [comment]
It fixes your particular test case, and it also has GLPK_MI passing three standard MILP test cases. We can close this issue once #910 is merged, if you'd like. I'm also open to keeping this issue standing until a little longer after the PR (in case you'd like to try additional test-cases).

**Comment by wkschwartz** [comment]
I won’t be able to do any further testing for several months. Go ahead and close the ticket once merged. If I run into problems again, I’ll either comment on this ticket or open a new one, as appropriate. Thanks for taking care of this!

**Comment by rileyjmurray** [comment]
Sounds good! Thanks for the reply.

**Comment by rileyjmurray** [comment]
Closed by #910
